src/ert/resources/workflows/jobs/internal-gui/scripts/csv_export.py

```diff
@@ -57,59 +58,64 @@ def run(
-
             try:
-                ensemble = self.storage.get_ensemble_by_name(ensemble)
-            except KeyError as exc:
-                raise UserWarning(f"The ensemble '{ensemble}' does not exist!") from exc
-
-            if not ensemble.has_data():
-                raise UserWarning(f"The ensemble '{ensemble}' does not have any data!")
-
-            ensemble_data = ensemble.load_all_gen_kw_data()
+                if not ensemble.has_data():
+                    raise UserWarning(
+                        f"The ensemble '{ensemble.name}' does not have any data!"
+                    )
+
+                ensemble_data = ensemble.load_all_gen_kw_data()
+
+                if design_matrix_path is not None:
+                    design_matrix_data = loadDesignMatrix(design_matrix_path)
+                    if not design_matrix_data.empty:
+                        ensemble_data = ensemble_data.join(
+                            design_matrix_data, how="outer"
+                        )
+
+                misfit_data = facade.load_all_misfit_data(ensemble)
+                if not misfit_data.empty:
+                    ensemble_data = ensemble_data.join(misfit_data, how="outer")
+
+                summary_data = ensemble.load_all_summary_data()
+                if not summary_data.empty:
+                    ensemble_data = ensemble_data.join(summary_data, how="outer")
+                else:
+                    ensemble_data["Date"] = None
+                    ensemble_data.set_index(["Date"], append=True, inplace=True)
+
+                ensemble_data["Iteration"] = ensemble.iteration
+                ensemble_data["Ensemble"] = ensemble.name
+                ensemble_data.set_index(
+                    ["Ensemble", "Iteration"], append=True, inplace=True
+                )
+
+                data = pandas.concat([data, ensemble_data])
 
-            if design_matrix_path is not None:
-                design_matrix_data = loadDesignMatrix(design_matrix_path)
-                if not design_matrix_data.empty:
-                    ensemble_data = ensemble_data.join(design_matrix_data, how="outer")
-
-            misfit_data = facade.load_all_misfit_data(ensemble)
-            if not misfit_data.empty:
-                ensemble_data = ensemble_data.join(misfit_data, how="outer")
-
-            summary_data = ensemble.load_all_summary_data()
-            if not summary_data.empty:
-                ensemble_data = ensemble_data.join(summary_data, how="outer")
-            else:
-                ensemble_data["Date"] = None
-                ensemble_data.set_index(["Date"], append=True, inplace=True)
-
-            ensemble_data["Iteration"] = ensemble.iteration
-            ensemble_data["Ensemble"] = ensemble.name
-            ensemble_data.set_index(
-                ["Ensemble", "Iteration"], append=True, inplace=True
-            )
-
-            data = pandas.concat([data, ensemble_data])
+            except KeyError as exc:
```

Comment:
Not sure where this exception belongs to? Maybe where the ensemble is loaded: `ensemble = self.storage.get_ensemble(ensemble_id)` since previously:
```python
 try:
                ensemble = self.storage.get_ensemble_by_name(ensemble)
            except KeyError as exc:
                raise UserWarning(f"The ensemble '{ensemble}' does not exist!") from exc
```